src/build.py

```diff
@@ -1130,99 +1134,112 @@ def DebianPackage():
+        out=outsubdir,
+        config='wasm-' + extension,
+        opt=opt)
+    UploadArchive('torture-%s-%s' % (extension, opt), Archive(outsubdir))
+    if 0 != unexpected_result_count:
+      buildbot.Fail()
 
 
 def CompileLLVMTortureBinaryen(name, em_config, outdir, fails):
-  buildbot.Step('Compile LLVM Torture (%s)' % name)
-  os.environ['EM_CONFIG'] = em_config
-  c = Executable(os.path.join(INSTALL_DIR, 'emscripten', 'emcc'), '.bat')
-  cxx = Executable(os.path.join(INSTALL_DIR, 'emscripten', 'em++'), '.bat')
-  Remove(outdir)
-  Mkdir(outdir)
-  unexpected_result_count = compile_torture_tests.run(
-      c=c, cxx=cxx, testsuite=GCC_TEST_DIR,
-      sysroot_dir=INSTALL_SYSROOT,
-      fails=fails,
-      out=outdir,
-      config='binaryen-native')
-  UploadArchive('torture-' + name, Archive(outdir))
-  if 0 != unexpected_result_count:
-    buildbot.Fail()
+  for opt in BINARYEN_TEST_OPT_FLAGS:
+    buildbot.Step('Compile LLVM Torture (%s, %s)' % (name, opt))
+    os.environ['EM_CONFIG'] = em_config
+    c = Executable(os.path.join(INSTALL_DIR, 'emscripten', 'emcc'), '.bat')
+    cxx = Executable(os.path.join(INSTALL_DIR, 'emscripten', 'em++'), '.bat')
+    outsubdir = os.path.join(outdir, opt)
+    Remove(outsubdir)
+    Mkdir(outsubdir)
+    unexpected_result_count = compile_torture_tests.run(
+        c=c, cxx=cxx, testsuite=GCC_TEST_DIR,
+        sysroot_dir=INSTALL_SYSROOT,
+        fails=fails,
+        out=outsubdir,
+        config='binaryen-native',
+        opt=opt)
+    UploadArchive('torture-%s-%s' % (name, opt), Archive(outsubdir))
+    if 0 != unexpected_result_count:
+      buildbot.Fail()
   return outdir
 
 
-def LinkLLVMTorture(name, linker, fails, input_pattern, args=None):
-  buildbot.Step('Link LLVM Torture with %s' % name)
-  assert os.path.isfile(linker), 'Cannot find linker at %s' % linker
-  out = os.path.join(WORK_DIR, 'torture-%s' % name)
-  Remove(out)
-  Mkdir(out)
-  unexpected_result_count = link_assembly_files.run(
-      linker=linker, files=input_pattern, fails=fails, out=out, args=args)
-  UploadArchive('torture-%s' % name, Archive(out))
-  if 0 != unexpected_result_count:
-    buildbot.Fail()
-  return out
+def LinkLLVMTorture(name, linker, fails, indir, extension, args=None):
+  for opt in BARE_TEST_OPT_FLAGS:
+    buildbot.Step('Link LLVM Torture (%s, %s)' % (name, opt))
+    assert os.path.isfile(linker), 'Cannot find linker at %s' % linker
+    outsubdir = os.path.join(WORK_DIR, 'torture-%s' % name, opt)
+    Remove(outsubdir)
+    Mkdir(outsubdir)
+    input_pattern = os.path.join(indir, opt, '*.' + extension)
+    unexpected_result_count = link_assembly_files.run(
+        linker=linker, files=input_pattern, fails=fails, attributes=[opt],
+        out=outsubdir, args=args)
+    UploadArchive('torture-%s-%s' % (name, opt), Archive(outsubdir))
+    if 0 != unexpected_result_count:
+      buildbot.Fail()
+  return os.path.join(WORK_DIR, 'torture-%s' % name)
```

Comment:
Nit: Instead of repeating `os.path.join(WORK_DIR, 'torture-%s' % name)`, we can create a temporary `out` and `os.path.join(out, opt)`, which is slightly-better-documenting what that expression means (and similarly explains how `outsubdir` is a subdir)